### PR TITLE
Feature/geinfra 341

### PR DESCRIPTION
**Pull request by AltusBarry**
Added DeletionMethod API endpoints and tests. Moved Site field validation to model __init__ to guard against cases where full model data is not available in the validation method.

### CODE REVIEW

File: swagger/access_control.yml [hunk 1]
```diff
@@ -871,6 +877,61 @@ definitions:
...
+    type: object
+    properties:
+      label:
+        type: string
+        maxLength: 100
+      data_schema:
+        type: object
+      description:
+        type: string
+
```
Comment (cobusc) [inline swagger/access_control.yml]: Add `minProperties: 1`, please.

File: swagger_server/models/site_create.py [hunk 2]
```diff
@@ -201,6 +201,8 @@ def deletion_method_id(self, deletion_method_id: int):
         :param deletion_method_id: The deletion_method_id of this SiteCreate.
         :type deletion_method_id: int
         """
+        if deletion_method_id is None:
```
Comment (cobusc) [inline swagger_server/models/site_create.py]: This is generated code. I'd prefer that this type of logic is performed in the controller.

File: swagger_server/models/site_create.py [hunk 3]
```diff
@@ -222,5 +224,7 @@ def deletion_method_data(self, deletion_method_data: object):
         :param deletion_method_data: The deletion_method_data of this SiteCreate.
         :type deletion_method_data: object
         """
+        if deletion_method_data is None:
```
Comment (cobusc) [inline swagger_server/models/site_create.py]: This is generated code. I'd prefer that this type of logic is performed in the controller.

File: swagger_server/test/controllers/test_site_controller.py [hunk 4]
```diff
@@ -48,13 +48,14 @@ def setUp(self):
     def test_site_create(self):
         """Test case for site_create
         """
-        data = Site(**{
-            "name": ("%s" % uuid.uuid1())[:30],
-            "domain_id": self.domain_model.id,
```
Comment (cobusc) [inline swagger_server/test/controllers/test_site_controller.py]: Is `domain_id` not required when creating a site?

File: swagger_server/test/controllers/test_site_controller.py [hunk 5]
```diff
@@ -48,13 +48,14 @@ def setUp(self):
...
+            "deletion_method_data": {},
+            "deletion_method_id": 0,
+            "description": "a super cool test site",
+            "domain_id": self.domain_model.id,
             "is_active": True,
-            "deletion_method_id": 0
+            "name": ("%s" % uuid.uuid1())[:30],
         })
         response = self.client.open(
             '/api/v1/sites',
```
Comment (cobusc) [inline swagger_server/test/controllers/test_site_controller.py]: Can you add a test where the `deletion_method_id` is not `0`, i.e. a test where the deletion method data will be validated against a proper schema?

File: swagger_server/models/site_create.py (hunk 2 above)
Comment (AltusBarry) [reply swagger_server/models/site_create.py]: This was generated, seems it was missing during last generation

File: swagger_server/models/site_create.py (hunk 3 above)
Comment (AltusBarry) [reply swagger_server/models/site_create.py]: This was generated, seems it was missing during last generation

File: swagger_server/test/controllers/test_site_controller.py (hunk 5 above)
Comment (AltusBarry) [reply swagger_server/test/controllers/test_site_controller.py]: https://github.com/girleffect/core-access-control/blob/develop/access_control/test/test_model_validation.py#L15

File: swagger/access_control.yml (hunk 1 above)
Comment (AltusBarry) [reply swagger/access_control.yml]: Added

**Comment by cobusc** [comment]
:+1:

**Pull request closed without merging**